doc/btp_has.txt

```diff
@@ -0,0 +1,107 @@
+HAS Service (ID 15)
+===================
+
+Commands and responses:
+
+	Opcode 0x00 - Error response
+
+	Opcode 0x01 - Read Supported Commands command/response
+
+		Controller Index:	 <controller id>
+		Command parameters:	 <none>
+		Response parameters: <supported commands> (variable)
+
+		Each bit in response is a flag indicating if command with
+		opcode matching bit number is supported. Bit set to 1 means
+		that command is supported. Bit 0 is reserved and shall always
+		be set to 0. If specific bit is not present in response (less
+		than required bytes received) it shall be assumed that command
+		is not supported.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x02: HAS Set Active Preset Index
+
+		Controller Index:	 <controller id>
+		Command parameters:	 Address_Type (1 octet)
+							 Address (6 octets)
+						 	 Preset Index (1 octet)
+		Response parameters: <None>
+
+		This command is used to set the Active Preset. 
+		If called with a Preset Index of 0, it will de-select the Active Preset.
+		
+		In case of an error, the error response will be returned.
+
+	Opcode 0x03: HAS Set Preset Name
+
+		Controller Index:	 <controller id>
+		Command parameters:	 Address_Type (1 octet)
+							 Address (6 octets)
+							 Preset Index (1 octet)
+							 Name Length (1 octet)
+							 Name (<Name Length> octets)
+		Response parameters: <None>
+
+        Name shall be a UTF-8 encoded string. 
+		Name Length holds the number of bytes in the UTF-8 encoded name. 
+
+		This command is used to change the Name of a Preset.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x04: HAS Remove Preset
+
+		Controller Index:	 <controller id>
+		Command parameters:	 Address_Type (1 octet)
+							 Address (6 octets)
+							 Preset Index (1 octet)
+		Response parameters: <None>
+
+		Passing Preset Index := 0 will remove all active Presets.
+
+		This command is used to delete a specific Preset or all Presets.
+
+		In case of an error, the error response will be returned.
+
+	Opcode 0x05: HAS Add Preset
+
+		Controller Index:	 <controller id>
+		Command parameters:	 Address_Type (1 octet)
+							 Address (6 octets)
+							 Preset Index (1 octet)
+							 Preset Properties (1 octet)
+							 Name Length (1 octet)
+							 Name (<Name Length> octets)
+		Response parameters: <None>
+
+		Preset Properties is a bit-field, with the following interpretation: 
+		Preset Properties:
+				1 - Preset will be Writable (N.U.)
+				2 - Preset will be Available
+
+        Name shall be a UTF-8 encoded string. 
```

Comment:
clarify if string should be null terminated or not (I guess null is not needed since we have length param)